### PR TITLE
feat(core, windows): Add api to access firebase cpp objects

### PR DESCRIPTION
**Pull request by cynthiajoan**
## Description

Add apis to firebase_core windows to access firebase cpp object like App*, Auth* etc.

## Related Issues

N/A

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_core/firebase_core/windows/firebase_core_plugin.cpp [hunk 1]
```diff
@@ -38,6 +42,26 @@ void FirebaseCorePlugin::RegisterWithRegistrar(
   registrar->AddPlugin(std::move(plugin));
 }
 
+void *FirebaseCorePlugin::GetFirebaseApp(std::string appName) {
+  return App::GetInstance(appName.c_str());
+}
+
+void *FirebaseCorePlugin::GetFirebaseAuth(std::string appName) {
```
Comment (Lyokone) [inline packages/firebase_core/firebase_core/windows/firebase_core_plugin.cpp:49]: It makes me think that auth will probably need a `tenantId` parameter but I don't think this is available yet on C++.

**Review by Lyokone — APPROVED** [review]

**Review by russellwheatley — APPROVED** [review]

File: packages/firebase_core/firebase_core/windows/CMakeLists.txt [hunk 2]
```diff
@@ -90,8 +90,8 @@ set(MSVC_RUNTIME_MODE MD)
 add_subdirectory(${FIREBASE_CPP_SDK_DIR} bin/ EXCLUDE_FROM_ALL)
 target_include_directories(${PLUGIN_NAME} INTERFACE
   "${FIREBASE_CPP_SDK_DIR}/include")
-set(firebase_libs firebase_auth firebase_database firebase_app)
-foreach(firebase_lib IN ITEMS ${firebase_libs})
+set(FIREBASE_LIBS firebase_app firebase_auth firebase_remote_config)
```
[the remark this answers is not in the document]
Comment (cynthiajoan) [reply packages/firebase_core/firebase_core/windows/CMakeLists.txt:93]: My understanding of cmake link sequence is the lib that is dependent by other libs should be linked first. Eg auth depends on app, so should link app before auth.

File: packages/firebase_core/firebase_core/windows/firebase_core_plugin.cpp (hunk 1 above)
Comment (cynthiajoan) [reply packages/firebase_core/firebase_core/windows/firebase_core_plugin.cpp:49]: what is the purpose of tenantId? I'd like to see how to map that into the c++ lib.

Comment (Lyokone) [reply packages/firebase_core/firebase_core/windows/firebase_core_plugin.cpp:49]: It is sued to determine the tenant of the app: https://cloud.google.com/identity-platform/docs/multi-tenancy

**Pull request merged**